### ISSUE TITLE
USWDS - Enhancement: Tooltip needs to announce "tooltip" for context

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds#5891]
### Describe the bug

The [tooltip component ](https://designsystem.digital.gov/components/tooltip/) the word "tooltip" isn't announced as a label.

The element that serves as the tooltip container should have a role [tooltip](https://w3c.github.io/aria/#tooltip).

Screen reader users are not able to discern the purpose of elements with role="tooltip" that do not have an accessible name. Aria tooltip elements must have discernible text that clearly describes the destination, purpose, function, or action for screen reader users.

[From w3c](https://w3c.github.io/aria/#tooltip): Authors SHOULD ensure that elements with the [role](https://w3c.github.io/aria/#dfn-role) tooltip are referenced through the use of [aria-describedby](https://w3c.github.io/aria/#aria-describedby) before or at the time the tooltip is displayed.

### Steps to reproduce the bug

1. Go to the tooltip component page
2. Listen to the example buttons using assistive technology (in my case I used JAWS and NVDA)
3. You hear the tooltip wording announced ("show on top button; top") but it's not declaring it as a tooltip. Currently it sounds like the button is repeating itself.

### Expected Behavior

Listen to the [tooltip example](https://dequeuniversity.com/library/aria/tooltip) on Deque's website. You can hear "tooltip colon" then the tooltip wording.

Ideally on our component examples you would hear "Tooltip colon: Top. Show on top".

### System setup

PC, Chrome, JAWS, NVDA

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct. 
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate bug reports.

**Comment by mejiaj** [comment]
Nice find, @amyleadem. 

Enhancement seems correct if the correct role is currently included & the description/trigger are linked to each other.

**Comment by amycole501** [comment]
Great information, Amy! I agree then, that this can be considered an enhancement. Albeit, the tooltip to me still sounded really confusing without the enhancement. The Deque example was much clearer. This would be a good example of something to test in our next round of usability research for conformance which is above and beyond compliance.

**Comment by alex-hull** [comment]
I think we consider marking this as conditional? Since we're not marking it as fail, I also do not think it would fall into the passes with exceptions category.

**Comment by amycole501** [comment]
I think conditional is correct; the accessible name of a tooltip can come from the content. So adding the ARIA properties to provide a tooltip with an accessible name is not recommended.  https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role

**Comment by amyleadem** [comment]
If we mark these as conditional with the tests as written, I worry that users will go looking for the word "tooltip" in their project readouts but won't find it. 

If this is correctly coded according to the [w3c standards](https://w3c.github.io/aria/#tooltip) (namely, aria-describedby), it just means that the tooltip content will be read by the screen reader when you focus on the tooltip trigger. Could the test for 4.1.2 and/or 1.31 just be re-written to be something like: "the screen reader announces the tooltip content"?

**Comment by amycole501** [comment]
Yes I think that would work.

**Comment by alex-hull** [comment]
So marking it as conditional and then simplifying the gherkin? I think that would work as well!

**Comment by mejiaj** [comment]
@jaclinec could you help us with the following:

1. If `tooltips` is the most useful word to use when describing this component. 
2. List some tooltip component examples so we can see if they announce themselves as tooltips.

**Comment by jaclinec** [comment]
After discussing at CS UX sync today, the team agreed that we should set the aria role = tooltip so that it announces "tooltip" to screen reader users. 

We'll try to test this in the upcoming round of usability testing this winter.

**Comment by mlloydbixal** [comment]
I was looking at this again this morning and confirmed the code _does already_ use `role="tooltip"` correctly. In Amy's first example from deque in the issue description, the screen reader that is calling out "tooltip, colon" is reading an additional span inside the tool tip with that wording. We're also already using `aria-labeledby` correctly. I'm wondering if this means it's not a native functionality for the screen reader to announce "tooltip" ? Should we follow a similar solution to adding an extra span, hidden text element, or aria-label that should be announced before the tooltip's text content?

cc @mejiaj @amycole501 @jaclinec

**Comment by amycole501** [comment]
> I was looking at this again this morning and confirmed the code _does already_ use `role="tooltip"` correctly. In Amy's first example from deque in the issue description, the screen reader that is calling out "tooltip, colon" is reading an additional span inside the tool tip with that wording. We're also already using `aria-labeledby` correctly. I'm wondering if this means it's not a native functionality for the screen reader to announce "tooltip" ? Should we follow a similar solution to adding an extra span, hidden text element, or aria-label that should be announced before the tooltip's text content?
> 
> cc [@mejiaj](https://github.com/mejiaj) [@amycole501](https://github.com/amycole501) [@jaclinec](https://github.com/jaclinec)

Yes I'd like to test the span solution since it worked well on Deque's example.

**Comment by mejiaj** [comment]
The proposed feature was added to uswds/uswds-sandbox#168.

We're going to hold on this feature until **after** user testing has finished.

**Comment by dct2023** [comment]
@mejiaj Should this be moved to the next sprint and assigned out or will this be on hold for a longer period of time?

**Comment by jaclinec** [comment]
@dct2023 testing is slated for this and next sprint, so my hunch is that this will be on hold until the sprint after next. So, can probably be moved to 2025.4

**Comment by dct2023** [comment]
@jaclinec Is the testing being tracked in another issue?

**Comment by jaclinec** [comment]
@dct2023 yes it's this epic in the uswds-team repo: [#436](https://github.com/uswds/uswds-team/issues/436)

**Comment by dct2023** [comment]
@jaclinec Thank you for the note! Do we need it to be moved to the next sprint and assigned to you? Is this part of a deliverable for February or March?

cc: @mejiaj @jdotyoon

**Comment by amycole501** [comment]
This was undergoing testing for confirmation with UX research. Will update with findings as UX research data is reviewed.

**Comment by mgifford** [comment]
- I've collected some tooltip examples here:

https://mgifford.github.io/ACCESSIBILITY.md/examples/TOOLTIP_ACCESSIBILITY_BEST_PRACTICES.html

It is important to add the role [tooltip](https://w3c.github.io/aria/#tooltip) as stated in the description.

The current approach is essentially a title tag applied to a button. Buttons could be an example, but to demonstrate "A tooltip is a short descriptive message that appears when a user hovers or focuses on an element." we do need to show more than the button element. 

Current code is:

```
    <button
      type="button"
      class="usa-button usa-tooltip"
      data-position="top"
      title="Top"
    >
      Show on top
    </button>
```

Worth looking at:

- https://design.cms.gov/components/tooltip/?theme=core
- https://www.telerik.com/design-system/docs/components/tooltip/
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tooltip_role
- https://www.a11y-collective.com/blog/tooltips-in-web-accessibility/
- https://www.digitala11y.com/tooltip-role/
- https://accessibilityinsights.io/info-examples/web/aria-tooltip-name/
- https://techservicesillinois.github.io/accessibility/aria-examples/tooltip.html

**Comment by ryparker** [comment]
I'm keeping this open as an enhancement, consistent with the earlier investigation. The tooltip already has its role and an association with its trigger; the proposed addition is saying “tooltip” before the content.

The `Status: Not a Bug` label reflects that distinction. The proposed improvement still needs evaluation, using the usability research and existing PRs.

**Comment by ryparker** [comment]
We're keeping this open to find out whether saying "tooltip" before its content helps screen-reader users understand it. USWDS already adds `role="tooltip"` and connects the tooltip to the element that opens it. The proposed change is the additional spoken word.

Before choosing an implementation, we need the results of the planned user testing, plus checks with current screen readers and translated text. If adding "tooltip" improves understanding, we should choose one implementation. If it does not, we should keep the current behavior.
